airflow/api_fastapi/core_api/datamodels/monitor.py

```diff
@@ -49,4 +49,4 @@ class HealthInfoSchema(BaseModel):
     metadatabase: BaseInfoSchema
     scheduler: SchedulerInfoSchema
     triggerer: TriggererInfoSchema
-    dag_processor: DagProcessorInfoSchema
+    dag_processor: DagProcessorInfoSchema | None = None
```

Comment:
I don't think this can be none. It can be 'unset' or DagProcessorInfoSchema, but `none` should not be possible as I understand.